### PR TITLE
Enable shell check scripts for existing scripts

### PR DESCRIPTION
**Pull request by spyridon97**
Enable shell check scripts for existing scripts

### CODE REVIEW

**Comment by vicentebolea** [comment]
@spyridon97 thanks for the this change. We have to be very careful with these changes since we do not test all those files in CI, that it is why we have this "exclude" list. Everything which is not cover in CI should not be modified (Unless you run those scripts and verify that they work). 

Please remove changes for `scripts/runconf` `scripts/docker` `scripts/developer/` `scripts/ci/scripts/`

File: scripts/ci/gh-actions/macos-setup.sh [hunk 1]
```diff
@@ -8,13 +8,13 @@ then
   echo "Error: GH_YML_MATRIX_COMPILER variable is not defined"
   exit 1
 fi
-XCODE_VER="$(echo ${GH_YML_MATRIX_COMPILER} | sed -e 's|_|.|g' -e 's|xcode||')"
-if [ ! -d /Applications/Xcode_${XCODE_VER}.app ]
+XCODE_VER="$(echo "${GH_YML_MATRIX_COMPILER}" | sed -e 's|_|.|g' -e 's|xcode||')"
+if [ ! -d /Applications/Xcode_"${XCODE_VER}".app ]
```
Comment (vicentebolea) [inline scripts/ci/gh-actions/macos-setup.sh:12]:
```suggestion
if [ ! -d "/Applications/Xcode_${XCODE_VER}.app" ]
```

File: scripts/ci/gh-actions/macos-setup.sh [hunk 2]
```diff
@@ -8,13 +8,13 @@ then
...
 then
   echo "Error: XCode installation directory /Applications/Xcode_${XCODE_VER}.app does not exist"
   exit 2
 fi
-sudo xcode-select --switch /Applications/Xcode_${XCODE_VER}.app
+sudo xcode-select --switch /Applications/Xcode_"${XCODE_VER}".app
```
Comment (vicentebolea) [inline scripts/ci/gh-actions/macos-setup.sh:17]: same

File: scripts/ci/gh-actions/run.sh [hunk 3]
```diff
@@ -100,19 +100,19 @@ export ADIOS2_IP=127.0.0.1
 export OMP_NUM_THREADS=1
 
 # Load any additional setup scripts
-if [ -f gha/scripts/ci/setup-run/ci-${GH_YML_JOBNAME}.sh ]
+if [ -f gha/scripts/ci/setup-run/ci-"${GH_YML_JOBNAME}".sh ]
```
Comment (vicentebolea) [inline scripts/ci/gh-actions/run.sh:103]: same with the quotes in this file

**Comment by spyridon97** [comment]
@vicentebolea i see a weird ascent kokkos failure, shall i ignore this? If yes, can you approve?

File: scripts/ci/circle/run.sh [hunk 4]
```diff
@@ -8,8 +9,9 @@ REALBRANCH="${CIRCLE_BRANCH}"
 if [ -n "${CIRCLE_PR_NUMBER}" ]
 then
   APIURL="${API_BASE}/pulls/${CIRCLE_PR_NUMBER}"
-  RESULT="$(curl -s ${APIURL} | python3 -c "import sys, json; print(json.load(sys.stdin)['head']['ref'])" 2> /dev/null)"
-  if [ $? -eq 0 ]
+  RESULT="$(curl -s "${APIURL}" | python3 -c "import sys, json; print(json.load(sys.stdin)['head']['ref'])" 2> /dev/null)"
+  exit_status=$?
+  if [ "$exit_status" -eq 0 ];
```
Comment (vicentebolea) [inline scripts/ci/circle/run.sh:14]:
```suggestion
  if [ "$exit_status" -eq 0 ]
```

conditionals in bash does not need a semicolon if the `then` is in the next line

Apply this change in for the below cases too

**Review by vicentebolea — CHANGES_REQUESTED** [review]

**Comment by vicentebolea** [comment]
> @vicentebolea i see a weird ascent kokkos failure, shall i ignore this? If yes, can you approve?

Yeah, the ascent build is flaky, you can ignore it

**Comment by spyridon97** [comment]
@vicentebolea can you approve now?

**Review by vicentebolea — APPROVED** [review]

**Pull request merged**